### PR TITLE
docs: vignette

### PR DESCRIPTION
**Pull request by timcadman**
## To test
- Read readme
- Read through vignette
- Check English
- Check content

### CODE REVIEW

File: vignettes/dstidyverse.Rmd.orig [hunk 1]
```diff
@@ -26,111 +26,134 @@ options(knitr.duplicate.label = "allow")
...
-### Pattern matching
+### Rename
+`ds.rename` renames columns within a serverside dataframe. Column names are passed unquoted and are
+evaluated as column names within `df.name`. Below are some examples, see `dplyr::rename` for full details.
+
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(matches('[aeiou]')), newobj = "test")
-ds.colnames("test")
+ds.rename(df.name = "mtcars", tidy_select = list(effiency = mpg, power = hp), newobj = "new_df")
```
Comment (erikzwart) [inline vignettes/dstidyverse.Rmd.orig:113]: #> Error in str_sub(str_replace_all(string, pattern = fixed("list("), replacement = ""), : could not find function "str_sub"

File: vignettes/dstidyverse.Rmd.orig [hunk 2]
```diff
@@ -26,111 +26,134 @@ options(knitr.duplicate.label = "allow")
...
 dslite.server$assignMethod("selectDS", "selectDS")
+dslite.server$assignMethod("renameDS", "renameDS")
 conns <- datashield.login(logindata.dslite.cnsim[1, ], assign=TRUE)
 ```
 
-## The old way
-With ds.dataFrameSubset, you could only subset columns by their index. E.g.
+## Functions
+### Select
+`ds.select`, implements all the flexibility of `dplyr::select` in subsetting columns. This includes:
```
Comment (erikzwart) [inline vignettes/dstidyverse.Rmd.orig:66]: typo: sub setting or sub-setting

File: vignettes/dstidyverse.Rmd.orig [hunk 3]
```diff
@@ -26,111 +26,134 @@ options(knitr.duplicate.label = "allow")
...
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(ends_with('m')), newobj = "test")
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(starts_with('c') | ends_with('b')), newobj = "new_df")
+ds.colnames("new_df")
 ```
-### Pattern matching
+### Rename
+`ds.rename` renames columns within a serverside dataframe. Column names are passed unquoted and are
```
Comment (erikzwart) [inline vignettes/dstidyverse.Rmd.orig:108]: typo: server side or server-side, data frame or data-frame

File: vignettes/dstidyverse.Rmd.orig [hunk 4]
```diff
@@ -26,111 +26,134 @@ options(knitr.duplicate.label = "allow")
...
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(matches('[aeiou]')), newobj = "test")
-ds.colnames("test")
+ds.rename(df.name = "mtcars", tidy_select = list(effiency = mpg, power = hp), newobj = "new_df")
+ds.colnames("new_df")
 ```
-### Multiple conditions with '&'
+
+### Mutate
+`ds.mutate` creates new columns in a serverside dataframe. These new columns are normally transformations
```
Comment (erikzwart) [inline vignettes/dstidyverse.Rmd.orig:118]: typo: server side or server-side, data frame or data-frame

File: vignettes/dstidyverse.Rmd.orig [hunk 5]
```diff
@@ -26,111 +26,134 @@ options(knitr.duplicate.label = "allow")
...
+ds.mean("mtcars$cyl")
+ds.mean("new_df$mpg_trans")
+ds.mean("mtcars$hp")
+ds.mean("mtcars$drat")
+ds.mean("mtcars$qsec")
+ds.mean("new_df$new_var")
 ```
-### Multiple conditions with '|'
+
+#### Choose where the new columns are positioned in the dataframe with `.before` and `.after`:
```
Comment (erikzwart) [inline vignettes/dstidyverse.Rmd.orig:139]: typo: data frame or data-frame or r 'dataframe'

**Review by erikzwart — CHANGES_REQUESTED** [review]

File: vignettes/dstidyverse.Rmd.orig (hunk 1 above)
Comment (timcadman) [reply vignettes/dstidyverse.Rmd.orig:113]: Could you check for me what version of the `stringr` package you have installed? I might need to specify a minimum.

Comment (erikzwart) [reply vignettes/dstidyverse.Rmd.orig:109]: stringr 1.5.1

**Pull request merged**